tt_metal/impl/dispatch/kernels/cq_prefetch.cpp

```diff
@@ -316,41 +335,53 @@ uint32_t process_debug_cmd(uint32_t cmd_ptr) {
         WAYPOINT("!CHK");
         ASSERT(0);
     }
-
+#endif
     return cmd->debug.stride;
 }
 
-template<bool cmddat_wrap_enable>
+template<bool cmddat_wrap_enable, bool send_to_dispatch_master>
```

Comment:
What if the template took in the downstream_cb_end, noc_xy params and passed them through to write_downstream and acquire_pages (rather than the boolean to decide).  I think the inliner should handle making all of those constexpr, but I think doing this would make the code slightly cleaner and remove any doubt

Reply:
Latest changes forward all dispatch_s vs dispatch_d state through a constexpr struct. Logic is cleaned up: no `constexpr conditionals` in these functions, but the numb er of template parameters was large so I encapsulated them in a struct.